### PR TITLE
Pagination v 0 1

### PR DESCRIPTION
**Pull request by IliaStepanov**
I've tried to made an example of paging for User entity. For now, only on /user/all/${pageNumber}.
Also added several buttons to check "number of users per page" varieties.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -33,11 +33,10 @@ dependencies {
 
     compile group: 'org.mindrot', name: 'jbcrypt', version: '0.4'
     compile group: 'javax.servlet', name: 'javax.servlet-api', version: '4.0.1'
-    
+
```
Comment (vakhonin) [inline build.gradle:36]: ...

File: src/main/java/com/epam/lowcost/controller/UserController.java [hunk 2]
```diff
@@ -130,7 +149,7 @@ public String deleteUser(@ModelAttribute(value = "sessionUser") User sessionUser
             return "redirect:" + TICKETS + SELF;
         }
         model.addAttribute("message", userService.deleteUser(id));
-        return "users";
+        return "redirect:/user/all/1";
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/UserController.java]: ?

File: src/main/java/com/epam/lowcost/controller/UserController.java [hunk 3]
```diff
@@ -5,37 +5,56 @@
...
     public String getById(@ModelAttribute(value = "sessionUser") User sessionUser, @RequestParam long id, Model model) {
         if (!sessionUser.isAdmin()) {
             return "redirect:" + TICKETS + SELF;
         }
-        model.addAttribute("user", userService.getById(id));
+        List<User> user = new ArrayList<>();
+        user.add(userService.getById(id));
+        model.addAttribute("users", user);
         model.addAttribute("message", "Here is your User!");
         return "users";
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/UserController.java:59]: ?

File: src/main/java/com/epam/lowcost/service/implementations/UserServiceImpl.java [hunk 4]
```diff
@@ -48,4 +50,33 @@ public User verifyUser(String email, String password) {
...
+        int users = userDAO.countUsers();
+        int pagesNum;
+        if (users % usersByPage != 0) {
+            pagesNum = (users / usersByPage + 1);
+        } else {
+            pagesNum = (users / usersByPage);
+        }
+        if (pageId >= pagesNum) {
+            pageId = pagesNum;
+        }
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/service/implementations/UserServiceImpl.java:68]: difficult....

File: build.gradle (hunk 1 above)
Comment (IliaStepanov) [reply build.gradle:36]: Missed it.

File: src/main/java/com/epam/lowcost/controller/UserController.java (hunk 2 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/UserController.java]: I will change it to constants if everything else is ok. For now I used string for simplicity.

File: src/main/java/com/epam/lowcost/controller/UserController.java (hunk 3 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/UserController.java:59]: Object we put to model should be a Collection in order to be iterable in same <c:forEach/> as every other returns from UserController.  Could be changed to userService.getById returning List<User>.

File: src/main/java/com/epam/lowcost/service/implementations/UserServiceImpl.java (hunk 4 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/service/implementations/UserServiceImpl.java:68]: Here I check if pageId we got from view is bigger than our max number of pages we will show, if it is we prevent it.

File: src/main/java/com/epam/lowcost/controller/UserController.java (hunk 2 above)
Comment (vakhonin) [reply src/main/java/com/epam/lowcost/controller/UserController.java]: no, it is a bad practice. Doing whole task then make pull request

Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/UserController.java]: Changed redirects to use constants.

File: src/main/java/com/epam/lowcost/controller/UserController.java [hunk 5]
```diff
@@ -5,37 +5,56 @@
...
+@SessionAttributes({"sessionUser", "number"})
 public class UserController {
 
     @Autowired
     UserService userService;
 
 
-    @GetMapping(value = ALL)
-    public String getAllUsers(@ModelAttribute(value = "sessionUser") User sessionUser, Model model) {
+    @GetMapping(value = "all/{pageId}")
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/UserController.java]: ?

File: src/main/java/com/epam/lowcost/controller/UserController.java [hunk 6]
```diff
@@ -5,37 +5,56 @@
...
+
+        Map<String, Object> pageRepresentation = userService.getUsersByPage(pageId, usersByPage);
+
+        model.addAttribute("pagesNum", pageRepresentation.get("pagesNum"));
+        model.addAttribute("users", pageRepresentation.get("users"));
+        model.addAttribute("pageId", pageRepresentation.get("pageId"));
         return "users";
     }
 
+    @GetMapping(value = "/setUsersByPage")
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/UserController.java]: ?

File: src/main/webapp/WEB-INF/views/users.jsp [hunk 7]
```diff
@@ -14,53 +15,103 @@
...
+            </form></td>
+        </tr>
+    </c:forEach>
+
+</table>
+<br/>
 
-<c:forEach items="${users}" var="user">
-    <c:out value="${user.toString()}"/><br/>
+<a href="/user/all/${pageId-1}">Previous</a>
```
Comment (vakhonin) [inline src/main/webapp/WEB-INF/views/users.jsp]: constants

File: src/main/java/com/epam/lowcost/controller/UserController.java (hunk 5 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/UserController.java]: Added constants.

File: src/main/java/com/epam/lowcost/controller/UserController.java (hunk 6 above)
Comment (IliaStepanov) [reply src/main/java/com/epam/lowcost/controller/UserController.java]: Added constants.

File: src/main/webapp/WEB-INF/views/users.jsp (hunk 7 above)
Comment (IliaStepanov) [reply src/main/webapp/WEB-INF/views/users.jsp]: Added constants.

File: src/main/java/com/epam/lowcost/controller/UserController.java [hunk 8]
```diff
@@ -5,37 +5,56 @@
...
+        model.addAttribute("pagesNum", pageRepresentation.get("pagesNum"));
+        model.addAttribute("users", pageRepresentation.get("users"));
+        model.addAttribute("pageId", pageRepresentation.get("pageId"));
         return "users";
     }
 
+    @GetMapping(value = SET_USERS_BY_PAGE)
+    public String setUsersByPage(@RequestParam String number, Model model) {
+        model.addAttribute("number", Integer.parseInt(number));
+        return "redirect:" + USER + ALL + FIRST_PAGE;
```
Comment (vakhonin) [inline src/main/java/com/epam/lowcost/controller/UserController.java:46]: maybe number of page shouldnt be in the URL...

**Pull request merged**